File: pset2 - arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

//Main Function
int main(void)
{
    //Variables
    string text;
    int letter = 0;
    int word = 0;
    int sentence = 0;

    //Repeat until find a valid value
    do 
    {
        text = get_string("Text: ");
    }
    while (text == NULL);

    for (int i = 0; i < strlen(text); i++)
    {
        //Using ternario operator
        isalpha(text[i]) ? letter++ : 0;

        if ((i == 0 && text[i] != ' ') || 
        (i != strlen(text) - 1 && text[i] == ' ' && text[i + 1] != ' ')) 
        {
            word++;
        }

        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence++;
        }
    }

    //Get the result
    float l = (letter / (float) word) * 100;
    float s = (sentence / (float) word) * 100;
    int grade = round(0.0588 * l - 0.296 * s - 15.8);
    if (grade > 1 && grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: pset6 - python/readability/readability.py
import cs50
import math


# Main Function
def main():
    # Variables
    text = cs50.get_string("Text: ").strip()
    letter, word, sentence = 0, 0, 0

    for i in range(len(text)):
        if (text[i].isalpha()):
            letter += 1

        if ((i == 0 and text[i] != ' ') or (i != len(text) - 1 and text[i] == ' ' and text[i + 1] != ' ')):
            word += 1

        if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
            sentence += 1

    # Get the result
    l = (letter / word) * 100
    s = (sentence / word) * 100
    grade = round(0.0588 * l - 0.296 * s - 15.8)
    if (grade > 1 and grade < 16):
        print(f"Grade {grade}")
    elif (grade >= 16):
        print("Grade 16+")
    else:
        print("Before Grade 1")


if __name__ == "__main__":
    main()